### PR TITLE
Change to more realistic cpu and memory requests and limits

### PR DESCRIPTION
**Pull request by tom1299**
### Type of change
* Documentation

### Description
The values in the documentation of the cruise control deployment description should reflect more realistic values, see issue #3904 as well. This pull request includes the change.

### Checklist
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by kyguy** [comment]
Thanks for the PR @tom1299!

These numbers are certainly better than what we have. If I may ask, how were these new numbers determined and/or under what conditions were these numbers tested (e.g. how much load and how many brokers, partitions etc)? The numbers look good to me for a minimal cluster but I want to justify the new number changes to the maintainers.

**Comment by tom1299** [comment]
To be honest, I haven‘t tested these numbers. I studied the cruise-control configuration [here](https://github.com/linkedin/cruise-control/wiki/Configurations) and found that (besides load / brokers and partitions) memory and cpu consumption depends on the number of threads configured (default is 8 & 1, but probably configured much higher in big clusters). With a default stack size of ~1MB per thread I estimated the minimum memory setting.
But I can‘t provide any numbers that these settings will work for, I‘m afraid. I would need to do more research for that.

**Comment by tomncooper** [comment]
Thanks for this @tom1299, we definitely need more memory, I am not sure how mutli-threaded Cruise Control is but I think we should probably make the request 1 CPU as the model building can be quite a task.

File: documentation/modules/cruise-control/proc-deploying-cruise-control.adoc [hunk 1]
```diff
@@ -44,11 +44,11 @@ spec:
       # ...
     resources: <3>
       requests:
-        cpu: 200m
-        memory: 64Mi
-      limits:
         cpu: 500m
```
Comment (tomncooper) [inline documentation/modules/cruise-control/proc-deploying-cruise-control.adoc:47]: This should probably be 1

**Comment by ppatierno** [comment]
I agree with @tomncooper and @kyguy but to say that these numbers are more realistic we should have a kind of proof for it.

**Comment by tom1299** [comment]
Thank you for your feedback. I took a look at the [start script for CC](https://github.com/linkedin/cruise-control/blob/master/kafka-cruise-control-start.sh#L131) and if not otherwise specified the max. heap memory setting is `1Gb`. So I would think that a resource limit of `1Gb` is indeed a reasonable (although Stack size might actually increase the overall memory consumption). I also asked for help for some reasonable values on the CC glitter forum.

**Comment by tom1299** [comment]
I adjusted the values a little bit. I think that the values for memory `request 512M` and `limit 2G` are in line when using the default values provided in the CC start script (heap space size `xmx=1G`). If the java process exceeds this heap space (for example if the model needs a more space) a `OutOfMemory` exception should be thrown. That leaves ~`1G` for the stack, which actually could accommodate quiet a lot of threads running in parallel.
Regarding CPU, I have to admit that I did not find any means nor do I have runtime performance data to deduct a reasonable value.

File: documentation/modules/cruise-control/proc-deploying-cruise-control.adoc (hunk 1 above)
Comment (tom1299) [reply documentation/modules/cruise-control/proc-deploying-cruise-control.adoc]: I updated this value to 1 as you suggested

**Review by tomncooper — APPROVED** [review]

**Comment by tomncooper** [comment]
I think we need to do some more extensive testing. However, until we have the opportunity to do that I think these are much more sensible settings.

**Comment by tomncooper** [comment]
@tom1299 you will need to sign off your commit to pass the DCO check. If you look at the DCO bot details it will give you the commands to amend the commit.

**Review by kyguy — APPROVED** [review]
Sounds reasonable to me!

Thanks again for looking into this @tom1299!

**Review by ppatierno — APPROVED** [review]

**Review by scholzj — APPROVED** [review]

**Pull request merged**